From 7505531c5cc07ab6ac422a542fbb672612ea8088 Mon Sep 17 00:00:00 2001
From: William Schaub <wschaub@genesi-tech.com>
Date: Mon, 24 Jun 2013 19:32:24 -0500
Subject: [PATCH 2/2] libata-pmp: add back NO_SRST to sil3x26 quirks

---
 drivers/ata/libata-pmp.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/ata/libata-pmp.c b/drivers/ata/libata-pmp.c
index f257180..d8fca74 100644
--- a/drivers/ata/libata-pmp.c
+++ b/drivers/ata/libata-pmp.c
@@ -389,9 +389,10 @@ static void sata_pmp_quirks(struct ata_port *ap)
 			/* link reports offline after LPM */
 			link->flags |= ATA_LFLAG_NO_LPM;
 
-			/* Class code report is unreliable. */
+			/* Class code report is unreliable. and SRST times out
+             * under certain configurations */
 			if (link->pmp < 5)
-				link->flags |= ATA_LFLAG_ASSUME_ATA;
+				link->flags |= ATA_LFLAG_NO_SRST | ATA_LFLAG_ASSUME_ATA;
 
 			/* port 5 is for SEMB device and it doesn't like SRST */
 			if (link->pmp == 5)
-- 
1.7.9.5

